index.js

```diff
@@ -47,29 +55,26 @@ WBMQTT.prototype.init = function (config) {
 
 	// Init MQTT client
 
-	var mqttOptions = {
-		client_id: self.config.clientId,
-		will_flag: true,
-		will_topic: self.prefix + "/connected",
-		will_message: "0",
-		will_retain: true
-	};
+	if (self.config.user != "none" && self.config.password != "none") {
+		self.client = new mqtt(self.config.host, parseInt(self.config.port), self.config.user, self.config.password, self.config.clientId);
+	}
+	else{
```

Comment:
This can be done in one line. Passing "" as user and password means no authentication.